tests/rptest/services/redpanda.py

```diff
@@ -2567,12 +2567,8 @@ def set_cluster_config(self,
         if admin_client is None:
             admin_client = self._admin
 
-        patch_result = admin_client.patch_cluster_config(
-            upsert={
-                k: v
-                for k, v in values.items() if v is not None
-            },
-            remove=[k for k, v in values.items() if v is None])
+        patch_result = admin_client.patch_cluster_config(upsert=values,
+                                                         remove=[])
```

Comment:
>dict of property name to value. if value is None, key will be removed from cluster config

It seems like the previous behavior was done intentionally. Do you know if this is being used anywhere? Or does this new code still remove non-optional configs when they're set to None?

Reply:
I inserted this https://github.com/redpanda-data/redpanda/commit/63748f324543c5371936eb7a3d5859c2db75b1ac while adding testing for log_segment_ms, but at the time, I misunderstood how to set an optional property to null.
While changing this I did a check on the codebase, and log_segment_ms_test.py is the only file using this. I'll fix the comment

Reply:
tiered_storage_model_test was failing because it was using this interface to try to set manifest_size to nullopt, but the result was resetting it to the default value of 64_kb. it can be seen in the failed runs in the cluster-config.yaml file, where it has the 64_kb value

Reply:
how/why does `tests/rptest/tests/log_segment_ms_test.py` pass after this change? the value set (nullopt) is just ignored/clamped?

Reply:
looks like previously you could have used

```
self.rpk.cluster_config_set(
                    "cloud_storage_spillover_manifest_max_segments", "null")
```

to set a value to nullopt if we were to believe this test https://github.com/redpanda-data/redpanda/blob/3d45a0750f83095c0d5ee0a7d50d3538734229c8/tests/rptest/infinite_retention/infinite_retention_test.py#L528

Reply:
> how/why does tests/rptest/tests/log_segment_ms_test.py pass after this change? the value set (nullopt) is just ignored/clamped?

The default value of log_segment_ms is 2weeks so from the perspective of the test it's not time rolling 

> looks like previously you could have used

def. we have probably ~20 ways of doing it